### PR TITLE
Test for validating sail dates of products

### PR DESCRIPTION
**Pull request by dheerajsrinath**

### CODE REVIEW

File: src/test/java/com/training/api/tests/shorex/ShorexTests.java [hunk 1]
```diff
@@ -64,4 +68,26 @@ public void testValidateProductTypeName() {
                     .isEqualTo("Shore Excursion");
         }
     }
+
+    @Test
+    public void testValidateOfferingDate() throws ParseException {
+        String pattern = "yyyyMMdd";
+        String stringDate = "";
+        Date date;
+        Date searchDate = new SimpleDateFormat(pattern).parse("20200809");
```
Comment (PranavHanwante) [inline src/test/java/com/training/api/tests/shorex/ShorexTests.java]: directly write in line 90

File: src/test/java/com/training/api/tests/shorex/ShorexTests.java [hunk 2]
```diff
@@ -64,4 +68,26 @@ public void testValidateProductTypeName() {
...
+        response = new RestEngine().getResponse(map.get("shorexUrl"), headerMap);
```
Comment (PranavHanwante) [inline src/test/java/com/training/api/tests/shorex/ShorexTests.java]: directly write in line 80

File: src/test/java/com/training/api/tests/shorex/ShorexTests.java [hunk 3]
```diff
@@ -64,4 +68,26 @@ public void testValidateProductTypeName() {
...
+        com.training.pojos.shorex.validate.Response responseElement =
+                response.as(com.training.pojos.shorex.validate.Response.class);
+
+        Assertions.assertEquals(200, responseElement.getStatus(), "REQUEST UNSUCCESSFUL");
+
+        for (int index = 0; index < responseElement.getPayload().getProducts().size(); index++) {
+            stringDate = responseElement.getPayload().getProducts().get(index).getOffering().get(0).getOfferingDate();
```
Comment (PranavHanwante) [inline src/test/java/com/training/api/tests/shorex/ShorexTests.java:85]: directly write in line 86

File: src/test/java/com/training/api/tests/shorex/ShorexTests.java [hunk 4]
```diff
@@ -64,4 +68,26 @@ public void testValidateProductTypeName() {
...
+            date = new SimpleDateFormat(pattern).parse(stringDate);
```
Comment (PranavHanwante) [inline src/test/java/com/training/api/tests/shorex/ShorexTests.java]: directly write in line 88

File: src/test/java/com/training/api/tests/shorex/ShorexTests.java (hunk 1 above)
Comment (akhil7000) [inline src/test/java/com/training/api/tests/shorex/ShorexTests.java]: instead of directly hardcoding the date, lets get the date from `shorexUrl`

**Review by akhil7000 — CHANGES_REQUESTED** [review]

File: src/test/java/com/training/api/tests/shorex/ShorexTests.java [hunk 5]
```diff
@@ -64,4 +68,28 @@ public void testValidateProductTypeName() {
                     .isEqualTo("Shore Excursion");
         }
     }
+
+    @Test
+    public void testValidateOfferingDate() throws ParseException {
+
+        String pattern = "yyyyMMdd";
+        String stringDate = "";
+        Date searchDate = new SimpleDateFormat(pattern)
```
Comment (PranavHanwante) [inline src/test/java/com/training/api/tests/shorex/ShorexTests.java:77]: directly write in line 93

Comment (dheerajsrinath) [reply src/test/java/com/training/api/tests/shorex/ShorexTests.java:77]: @PranavHanwante wont that take more resource? it has to retrieve the string from map and perform manipulations to get date each time to get a date that doesn't change

File: src/test/java/com/training/api/tests/shorex/ShorexTests.java [hunk 6]
```diff
@@ -64,4 +68,27 @@ public void testValidateProductTypeName() {
                     .isEqualTo("Shore Excursion");
         }
     }
+
+    @Test
+    public void testValidateOfferingDate() throws ParseException {
+
+        String pattern = "yyyyMMdd";
+        String stringDate = "";
```
Comment (akhil7000) [inline src/test/java/com/training/api/tests/shorex/ShorexTests.java]: Lets make this name more meaningful - `offeringDate`

**Review by akhil7000 — CHANGES_REQUESTED** [review]

File: src/test/java/com/training/api/tests/shorex/ShorexTests.java [hunk 7]
```diff
@@ -64,4 +68,34 @@ public void testValidateProductTypeName() {
...
+                    .getProducts().get(productIndex)
+                    .getOffering().size(); offeringIndex++) {
+
+                offeringDate = responseElement.getPayload()
+                                .getProducts().get(productIndex)
+                                .getOffering().get(offeringIndex).getOfferingDate();
+
+                softAssertions.assertThat(new SimpleDateFormat(pattern).parse(offeringDate))
+                        .as("Sail date is before Search Date")
+                        .isAfterOrEqualsTo(searchDate);
```
Comment (akhil7000) [inline src/test/java/com/training/api/tests/shorex/ShorexTests.java]: This looks to be a little complex in readability, so lets, maybe before the two for loops store the products in a list by using this : 
`List<Products> products=responseElement.getPayload().getProducts()`
and then use this list in the for loops.

**Review by akhil7000 — CHANGES_REQUESTED** [review]

File: src/test/java/com/training/api/tests/shorex/ShorexTests.java [hunk 8]
```diff
@@ -64,4 +70,32 @@ public void testValidateProductTypeName() {
...
+
+        com.training.pojos.shorex.validate.Response responseElement =
+                new RestEngine().getResponse(map.get("shorexUrl"), headerMap).as(com.training.pojos.shorex.validate.Response.class);
+
+        Assertions.assertEquals(200, responseElement.getStatus(), "REQUEST UNSUCCESSFUL");
+
+        List<Products> products=responseElement.getPayload().getProducts();
+
+        for(Products product:products)
+        {
```
Comment (akhil7000) [inline src/test/java/com/training/api/tests/shorex/ShorexTests.java]: syntax not alligned

**Review by akhil7000 — CHANGES_REQUESTED** [review]

**Review by akhil7000 — APPROVED** [review]

**Pull request merged**